contracts/arbitrators/KlerosLiquidProxy.sol

```diff
@@ -0,0 +1,221 @@
+        require(
+            claimIndexToDisputeId[claimIndex] == 0,
+            "Dispute already created"
+        );
+        uint256 disputeId = klerosArbitrator.createDispute{value: msg.value}(
+            uint256(type(IClaimsManager.ArbitratorDecision).max),
+            klerosArbitratorExtraData
+        );
+        disputeIdToClaimIndex[disputeId] = claimIndex;
+        claimIndexToDisputeId[claimIndex] = disputeId;
+        emit CreatedDispute(claimIndex, claimant, disputeId);
+        emit Dispute(klerosArbitrator, disputeId, META_EVIDENCE_ID, claimIndex);
+        emit Evidence(klerosArbitrator, claimIndex, claimant, evidence);
+        claimsManager.createDispute(claimIndex);
+    }
+
+    function submitEvidenceToKlerosArbitrator(
+        uint256 claimIndex,
+        string calldata evidence
+    ) external override onlyClaimForwardedToKleros(claimIndex) {
+        require(
+            claimsManager.isManagerOrMediator(msg.sender),
+            "Sender cannot mediate"
+        );
+        emit SubmittedEvidenceToKlerosArbitrator(
+            claimIndex,
+            msg.sender,
+            evidence
+        );
+        emit Evidence(klerosArbitrator, claimIndex, msg.sender, evidence);
+    }
+
+    function appealKlerosArbitratorRuling(uint256 claimIndex)
+        external
+        payable
+        override
+        onlyClaimForwardedToKleros(claimIndex)
+    {
+        (, , address claimant, , , , ) = claimsManager.claims(claimIndex);
+        // Ruling options
+        // 0: Kleros refused to arbitrate or ruled that it's not appropriate to
+        // pay out the claim or the settlement. We allow both parties to appeal this.
+        // 1: Pay the claim. Only the mediator can appeal this.
+        // 2: Pay the settlement. Only the claimant can appeal this.
+        // We don't check the dispute status (if it's appealable), as the appeal() call
+        // below should revert in that case anyway.
+        if (msg.sender == claimant) {
+            require(
+                klerosArbitrator.currentRuling(
+                    claimIndexToDisputeId[claimIndex]
+                ) != 1,
+                "Ruling agrees with claimant"
+            );
+        } else if (claimsManager.isManagerOrMediator(msg.sender)) {
+            require(
+                klerosArbitrator.currentRuling(
+                    claimIndexToDisputeId[claimIndex]
+                ) != 2,
+                "Ruling agrees with mediator"
+            );
+        } else {
+            revert("Only parties can appeal");
+        }
+        emit AppealedKlerosArbitratorRuling(
+            claimIndex,
+            msg.sender,
+            claimIndexToDisputeId[claimIndex]
+        );
+        klerosArbitrator.appeal{value: msg.value}(
+            claimIndexToDisputeId[claimIndex],
+            klerosArbitratorExtraData // Unused in KlerosLiquid
+        );
+    }
+
+    function rule(uint256 disputeId, uint256 ruling) external override {
```

Comment:
Don't we need to check that this function is only called by the `Arbitrator` contract?

Reply:
Now that I think about it, this is probably being checked in `ClaimsManager.resolveDispute()`

Reply:
No you're right, this should require the sender to be KlerosLiquid

Reply:
https://github.com/api3dao/claims-manager/pull/27/commits/3cf8a15495986a63ab62555dd9cacd2acb09995c